src/WebJobs.Script.WebHost/SecretManager.cs

```diff
@@ -101,6 +104,45 @@ public virtual FunctionSecrets GetFunctionSecrets(string functionName)
             });
         }
 
+        /// <summary>
+        /// Iterate through all function secret files and remove any that don't correspond
+        /// to a function.
+        /// </summary>
+        /// <param name="rootScriptPath">The root function directory.</param>
+        public void PurgeOldFiles(string rootScriptPath)
+        {
+            if (!Directory.Exists(rootScriptPath))
+            {
+                return;
+            }
+
+            var functionLookup = Directory.EnumerateDirectories(rootScriptPath).ToLookup(p => Path.GetFileName(p), StringComparer.OrdinalIgnoreCase);
```

Comment:
It is important that we determine functions based on the presence of a folder, not whether we've identified a valid function from that folder. This ensures that we don't delete logs/secrets for functions that transition into/out of invalid unparsable states.